// components/Otp/Otp.js
/* eslint-disable */
import React  from 'react';
import PropTypes from 'prop-types'; // ES6


const propTypes = {

};

class Otp extends React.Component {

    constructor(props) {
        super(props);
        this.resendOTP = this.resendOTP.bind(this);

        this.state = {
            start:Date.now(),
            diff:0,
            minutes:0,
            seconds:0,
            duration : 60 * this.props.minutes,
            resend: false,
        }
        this.timerState = '';
        this.timerCount = '';
        this.count = 0;

    }

    componentDidMount (){
        window.clearInterval(this.timerCount);
        this.timerCount = setInterval(()=>{
            this.timer();
        },1000)

    }

    starttimer(){
        window.clearInterval(this.timerCount);
        this.timerCount = setInterval(()=>{
            this.timer();
        },1000)
    }


    componentWillUnmount() {
        window.clearInterval(this.timerCount);
    }

    timer () {

        let diff = this.state.diff;
        let minutes = this.state.minutes;
        let seconds = this.state.seconds;

        diff = this.state.duration - (((Date.now() -  this.state.start) / 1000) | 0);
        minutes = (diff / 60) | 0;
        seconds = (diff % 60) | 0;
        minutes = minutes < 10 ? "0" + minutes : minutes;
        seconds = seconds < 10 ? "0" + seconds : seconds;
        this.setState({minutes:minutes,seconds:seconds});

        if (diff <= 0) {
            this.setState({resend:true});
            window.clearInterval(this.timerCount);
        }
    }



    resendOTP(evt) {
        this.setState({start:Date.now(),minutes:0,seconds:0,duration: 60 * this.props.minutes});
        this.props.resendEvent();
        this.starttimer();
        evt.preventDefault();
    }

    render () {
        return(
            <div>
                <h3 className="errMsg" style={this.props.style.otpTimer}>{this.state.minutes} : {this.state.seconds}</h3>
                 <button className="resend" style={this.props.style.resendBtn} onClick={this.resendOTP}>Resend</button>
            </div>
        );
    }
}
Otp.propTypes = propTypes;

export default Otp;
